Source/IGListManagementPerformable.h

```diff
@@ -0,0 +1,270 @@
+
+ @param identifier An identifier that was used for registering either a class or nib.
+ @param indexPath The index path of the current object that can be used for any additional configuration specific to the presenting object.
+
+ @return A fully configured cell object.
+
+ @note If the presenting object does not have a registered Class or nib for this identifier, this method should error.
+ */
+- (__kindof UIView *)dequeueReusableCellWithReuseIdentifier:(NSString *)identifier forIndexPath:(NSIndexPath *)indexPath;
+
+/**
+ Request a fully initialized, reusable view of the type and kind registered for the identifier.
+
+ @param elementKind The kind of view to dequeue.
+ @param identifier An identifier that was used for registering either a Class or nib.
+ @param indexPath The index path of the current object that can be used for any additional configuration specific to the presenting object.
+
+ @return A fully configured view object.
+
+ @note If the presenting object does not have a registered Class or nib for this kind and identifier, then this method should error.
+ */
+- (__kindof UIView *)dequeueReusableSupplementaryViewOfKind:(NSString *)elementKind withReuseIdentifier:(NSString *)identifier forIndexPath:(NSIndexPath *)indexPath;
+
+/**
+ The distance the content is inset from the containing presenting object.
+ */
+- (UIEdgeInsets)contentInset;
+
+/**
+ The distance from the content origin to the origin of the presenting object.
+ */
+- (CGPoint)contentOffset;
+
+/**
+ Update the `contentOffset` property using an animation or not.
+
+ @param contentOffset The updated content offset value.
+ @param animated Whether or not to animate the change in `contentOffset`.
+ */
+- (void)setContentOffset:(CGPoint)contentOffset animated:(BOOL)animated;
+
+/**
+ The `delegate` and `dataSource` objects are intended to be used in a similar manner to that of the `UICollectionView`
+ delegate and data source objects. Implementations of this protocol will need to define their own protocols for the
+ `delegate` and `dataSource` objects. In general keep the following in mind:
+
+ - The `delegate` object should be used for customizing the behavior of the presenting object. Generally it will be responsible for responding to user interaction along with potentially layout customizations.
+ - The `dataSource` object is responsible for providing the data to the presenting object. This responsibility will generally include number of items and any views to be displayed.
+ */
+@property (nonatomic, weak, nullable) id delegate;
+@property (nonatomic, weak, nullable) id dataSource;
+
+/**
+ The view to display behind all content within the presenting object.
+ */
+@property (nonatomic, strong, nullable) UIView *backgroundView;
+
+/**
+ A vestige of the `UICollectionView` origins of `IGListKit`. Depending on the implementation of the presenting object, this may be ignored.
+ */
+@property (nonatomic, strong) id collectionViewLayout;
```

Comment:
this one is pretty specific to `UICollectionView` and i'm not sure if there is a way we can avoid having it in this protocol. :-\